libcextract/SymbolExternalizer.cpp

```diff
@@ -513,6 +513,33 @@ bool SymbolExternalizer::FunctionUpdater::Update_References_To_Symbol(Stmt *stmt
       expr->setDecl(NewSymbolDecl);
       replaced = true;
     }
+  } else if (DeclStmt::classof(stmt)) {
+    /* To handle symbol renames on situations like the sizeof below
+     *
+     * static char x[4];
+     *
+     * void f(void) {
+     *    char y[sizeof(x)];
+     * }
+     */
+    DeclStmt *dstmt = (DeclStmt *) stmt;
+    Decl *ddecl = dstmt->getSingleDecl();
```

Comment:
Perhaps we should do an assertion here to check if the type of this  `Decl` is an array of some sort by using `ddecl->getType()->isArrayType()` just to be sure that we are not overwriting something we shouldn't.